src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotContainTypeNames.cs

```diff
@@ -0,0 +1,152 @@
+            s_types.Add("float64");
+            s_types.Add("int16");
+            s_types.Add("int32");
+            s_types.Add("int64");
+            s_types.Add("uint16");
+            s_types.Add("uint32");
+            s_types.Add("uint64");
+            s_types.Add("intptr");
+            s_types.Add("uintptr");
+            s_types.Add("ptr");
+            s_types.Add("uptr");
+            s_types.Add("pointer");
+            s_types.Add("upointer");
+            s_types.Add("single");
+            s_types.Add("double");
+            s_types.Add("decimal");
+            s_types.Add("guid");
+            s_types.Add("object");
+            s_types.Add("obj");
+            s_types.Add("string");
+        }
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.RegisterSymbolAction(AnalyzeNamedType, SymbolKind.NamedType);
+            analysisContext.RegisterSymbolAction(AnalyzeProperty, SymbolKind.Property);
+            analysisContext.RegisterSymbolAction(AnalyzeMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeSymbol(ISymbol symbol, SymbolAnalysisContext context)
+        {
+            var typeName = symbol.Name;
+            //check if memeber contains type name 
+            var isTypeName = IsViolatingIdentifierName(typeName);
+            if (isTypeName)
+            {
+                var diagnostic = Diagnostic.Create(Rule, symbol.Locations[0], typeName);
+                context.ReportDiagnostic(diagnostic);
+            }
+        }
+
+        /// <summary> 
+        /// Retrieve the name of the class/struct/enum and store it in 
+        /// a userdefinedtypes hashset to ensure no member or parameter uses that userdefined typename 
+        /// Also verify that these members are not violating the CA1720 rule 
+        /// </summary> 
+        /// <param name="context"></param> 
+        private static void AnalyzeNamedType(SymbolAnalysisContext context)
+        {
+            AnalyzeSymbol(context.Symbol, context);
+        }
+
+        private static void AnalyzeProperty(SymbolAnalysisContext context)
+        {
+            AnalyzeSymbol(context.Symbol, context);
+            var parameters = ((IPropertySymbol)context.Symbol).Parameters;
+            foreach (var param in parameters)
+            {
+                AnalyzeSymbol(param, context);
+            }
+        }
+
+        private static void AnalyzeMethod(SymbolAnalysisContext context)
+        {
+            AnalyzeSymbol(context.Symbol, context);
+            var parameters = ((IMethodSymbol)context.Symbol).Parameters;
+            foreach (var param in parameters)
+            {
+                AnalyzeSymbol(param, context);
+            }
+        }
+        
+        private static bool IsViolatingIdentifierName(string identifier)
+        {
+            if (s_types.Contains(identifier))
+                return true;
+
+            return false;
```

Comment:
Also, this might has a high false positive rate (for a rule that already generate a fair amount of them). For example, `Mint` is a fine name but will be reported. We probably want to port `WordParser` from FxCop SDK and use that.